circuit-types/src/fixed_point.rs

```diff
@@ -47,14 +35,24 @@ lazy_static! {
     pub static ref TWO_TO_M: BigUint = BigUint::from(1u8) << DEFAULT_FP_PRECISION;
 
     /// The shift, converted to a scalar
-    pub static ref TWO_TO_M_SCALAR: Scalar = biguint_to_scalar(&TWO_TO_M);
+    pub static ref TWO_TO_M_SCALAR: ScalarField = biguint_to_scalar(&TWO_TO_M).inner();
 
     /// Compute the constant 2^-M (mod p), so that we may conveniently reduce after
     /// multiplications
-    pub static ref TWO_TO_NEG_M: Scalar = {
-        let two_to_m_scalar = biguint_to_scalar(&TWO_TO_M);
-        two_to_m_scalar.inverse()
-    };
+    pub static ref TWO_TO_NEG_M: ScalarField = TWO_TO_M_SCALAR.inverse().unwrap();
+}
+
+// -----------
+// | Helpers |
+// -----------
+
+/// Shift a given Scalar by M bits to the right
+pub fn shift_scalar_by_m(scalar: Scalar) -> Scalar {
```

Comment:
nit: could call this `right_shift_scalar_by_m`, to make the direction clear from the fn name?